token/token.go

```diff
@@ -246,8 +248,36 @@ func WriteToken(token string, name string, fingerprints []string, emails []strin
 	return nil
 }
 
+var (
+	lookupGPGCmd sync.Once
+	hasGPGCmd    bool
+)
+
+func hasGPG() bool {
+	lookupGPGCmd.Do(func() {
+		_, err := exec.LookPath("gpg")
+		hasGPGCmd = err == nil
+	})
+	return hasGPGCmd
+}
+
 // Decrypts encrypted []byte with any available pgp key
 func Decrypt(encoded []byte) ([]byte, error) {
+	if hasGPG() {
+		var stderr, stdout bytes.Buffer
+		cmd := exec.Command("gpg", "--decrypt")
```

Comment:
I'd prefer to use `gpg --quiet --decrypt --batch` and then catch on stderr. Here's what it looks like with a password-based key:

```
tschuy/projects/gotp → gpg --batch --quiet --decrypt < temp 
gpg: gpg-agent is not available in this session
gpg: can't query passphrase in batch mode
gpg: Invalid passphrase; please try again ...
gpg: can't query passphrase in batch mode
gpg: Invalid passphrase; please try again ...
gpg: can't query passphrase in batch mode
gpg: decryption failed: secret key not available
```

Reply:
my main thing here is that if you're using password based encryption with a local key and the key is in gpg-agent, without batch you get a bunch of error text:

```
tschuy/projects/gotp → gpg --quiet --decrypt < temp 

You need a passphrase to unlock the secret key for
user: "Evan Tschuy <evan.tschuy@coreos.com>"
4096-bit RSA key, ID 0x0C590A882B35D2A4, created 2017-01-25
         (subkey on main key ID 0xCE8111086BAD4E55)

[decrypted message here]
```